### PR TITLE
Implementation + Tests for i32.div_s and i32.div_u. Runtime Errors

### PR DESCRIPTION
**Pull request by nerodesu017**
### Pull Request Overview

This pull request:
- adds the `i32.div_s` operation
- adds test cases for the `i32.div_s` operation
- adds the `i32.div_u` operation
- adds test cases for the `i32.div_u` operation
- refactors the `invoke_func` and `function` functions to be able to return an error through the Result return type - to add runtime errors

### Testing Strategy

This pull request was tested locally on the added test case and all others.

### TODO or Help Wanted

This pull request still needs a review.

### Formatting

- [x] Ran `cargo fmt`
- [x] Ran `cargo check`
- [x] Ran `cargo build`

### Github Issue

This pull request partially implements #2

### Author

Signed-off-by: Popescu Adrian <adrian.popescu@oxidos.io>

### CODE REVIEW

File: src/execution/mod.rs [hunk 1]
```diff
@@ -209,18 +211,59 @@ impl<'b> RuntimeInstance<'b> {
...
+                }
+                // i32.div_s: [i32 i32] -> [i32]
+                0x6D => {
+                    let dividend: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+                    let divisor: i32 = stack.pop_value(ValType::NumType(NumType::I32)).into();
+
+                    if dividend == 0 {
+                        return Err(crate::Error::RuntimeError(
+                            crate::core::error::RuntimeError::DivideBy0,
+                        ));
```
Comment (george-cosma) [inline src/execution/mod.rs:229]:
```rust
return Err(crate::Error::RuntimeError(
    crate::core::error::RuntimeError::DivideBy0,
));
```

This syntax is a bit too verbose. I would import the two error enums directly. resultin in
```rust
return Err(RuntimeError(DivideBy0));
```

**Review by george-cosma — APPROVED** [review]

File: src/execution/mod.rs (hunk 1 above)
Comment (wucke13) [reply src/execution/mod.rs:229]: AFAIK in the diff this is still not resolved.

**Review by wucke13 — APPROVED** [review]
Some minor nitpicks, bug generally LGTM!

File: src/execution/mod.rs (hunk 1 above)
Comment (nerodesu017) [reply src/execution/mod.rs]: Now it should be fixed

**Review by valexandru — APPROVED** [review]

**Pull request merged**